storage/indexer.go

```diff
@@ -0,0 +1,11 @@
+package storage
+
+import (
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// Indexer indexes all execution receipts in a finalized container block based on
+// their reference block ID and the executor ID.
+type Indexer interface {
+	IndexReceipts(blockID flow.Identifier) error
+}
```

Comment:
⚠️ 
**Caution**: you are proliferating code from the consensus nodes, that is only temporary and soon going to be replaced. 

* nodes might want to index receipts at different stages. Consensus nodes will soon index on `OnBlockIncorporated`: see PR https://github.com/onflow/flow-go/pull/450, specifically [`Indexer`](https://github.com/onflow/flow-go/blob/70c922dad4b7fe304e55f7ddbe1d5647ea582bd2/engine/consensus/matching/indexer.go)

I am happy with `Indexer` being a general interface. However, is this is the case, then `Indexer` should not make any determination _when_ the receipts are indexed. 

Yurii's ongoing work is taking this even a step further: 
* please take a look at issue https://github.com/dapperlabs/flow-go/issues/5326 (specifically the generalized API `ExecutionReceipts`). With this implementation, Receipts will be indexed as soon as they are stored. 


#### suggestion:
1. lets implement a dedicated indexer for verification nodes: 
    * instead of moving `engine/consensus/matching/indexer.go → storage/badger/indexer.go` just create your own implementation. 
    * I believe for your purposes, you only need https://github.com/onflow/flow-go/blob/b4caf451b8b1bc79a5a5cd91a6b9d165cd39e03d/storage/badger/indexer.go#L45-L59
    * Consensus nodes can then adapt their implementation according to their needs 
    * This will allow the Consensus nodes and Verification nodes to independently decide 
2.  Please remove from the Indexer API description 

```suggestion
// Indexer indexes the provided execution receipts. Each Receipt is indexed by the block the receipt pertains to.
type Indexer interface {
	IndexReceipts(Receipts  []*ExecutionReceipt) error
}
```
Note that I changed the argument of IndexReceipts from the block ID to a slice of receipts. Thereby we eliminate the error case where the input  `blockID` is not known. For the verification node, the block, _including_ the payload, is actually already available at the point where we call the indexer, so it is straight forward to take the receipts from the payload directly

Reply:
Applied the suggestion in a chain of commits, but this one would reflect the gist of it: https://github.com/onflow/flow-go/pull/427/commits/8aead9a925ad17a9bd093da16f7ac5bf2f85a430